filament/backend/src/metal/MetalShaderCompiler.mm

```diff
@@ -225,4 +229,22 @@ bool isReady() const noexcept {
     mCallbackManager.setCallback(handler, callback, user);
 }
 
+UTILS_NOINLINE
+void MetalShaderCompiler::MetalFunctionBundle::validate() const {
+    if (UTILS_LIKELY(!std::holds_alternative<Error>(mPrograms))) {
```

Comment:
couldn't this just make use of `isValid`?  Should `None` also throw something?

Reply:
Good point about `None`, I updated the code to also throw an exception when it encounters None.